cmd/exoip/main.go

```diff
@@ -305,23 +305,34 @@ func main() {
 	}()
 
 	engine.UpdatePeers()
+
 	go func() {
-		// pings every interval our peers
-		i := 0
+		interval := time.Duration(5 * time.Minute)
+		var elapsed time.Duration
 		for {
-			i = (i + 1) % 100
-			if i == 0 {
-				if err := engine.UpdateNic(); err != nil {
-					exoip.Logger.Crit(err.Error())
-				}
+			time.Sleep(interval - elapsed)
+			start := time.Now()
 
-				engine.UpdatePeers()
+			engine.UpdatePeers()
+			// Check that we still own the EIP
+			if err := engine.UpdateNic(); err != nil {
+				exoip.Logger.Crit(err.Error())
 			}
+			elapsed = time.Now().Sub(start)
```

Comment:
With the verbose option, can you add a log with the elapsed time to perform the check.

Reply:
I'll need to add another layer to `Logger` for a _debug_ level, but agreed.